packages/types/src/codec/EnumType.spec.ts

```diff
@@ -2,132 +2,213 @@
+          '0x0134120000'
+        ).value.toString()
+      ).toEqual('4660'); // 0x1234 in decimal
+    });
 
-  it('decodes from a JSON input (mixed case)', () => {
-    expect(
-      new EnumType(
-        { Text, U32 },
-        { 'U32': 42 }
-      ).value.toString()
-    ).toEqual('42');
-  });
+    it('decodes from a JSON input (mixed case)', () => {
+      expect(
+        new Enum(
+          { Text, U32 },
+          { 'U32': 42 }
+        ).value.toString()
+      ).toEqual('42');
+    });
 
-  it('decodes from JSON string', () => {
-    expect(
-      new EnumType(
-        { Null, U32 },
-        'null'
-      ).type
-    ).toEqual('Null');
-  });
+    it('decodes from JSON string', () => {
+      expect(
+        new Enum(
+          { Null, U32 },
+          'null'
+        ).type
+      ).toEqual('Null');
+    });
 
-  it('stringifies with custom types', () => {
-    class A extends Null { }
-    class B extends Null { }
-    class C extends Null { }
-    class Test extends EnumType<A | B | C> {
-      constructor (value?: string, index?: number) {
-        super({
-          a: A,
-          b: B,
-          c: C
-        }, value, index);
+    it('stringifies with custom types', () => {
+      class A extends Null { }
+      class B extends Null { }
+      class C extends Null { }
+      class Test extends Enum {
+        constructor (value?: string, index?: number) {
+          super({
+            a: A,
+            b: B,
+            c: C
+          }, value, index);
+        }
       }
-    }
 
-    expect(new Test().toJSON()).toEqual({ a: null });
-  });
+      expect(new Test().toJSON()).toEqual({ a: null });
+    });
 
-  it('creates via with', () => {
-    class A extends Null { }
-    class B extends U32 { }
-    class C extends Null { }
-    const Test = EnumType.with({ A, B, C });
+    it('creates via with', () => {
+      class A extends Null { }
+      class B extends U32 { }
+      class C extends Null { }
+      const Test = Enum.with({ A, B, C });
 
-    expect(new Test().toJSON()).toEqual({ A: null });
-    expect(new Test(1234, 1).toJSON()).toEqual({ B: 1234 });
-  });
+      expect(new Test().toJSON()).toEqual({ A: null });
+      expect(new Test(1234, 1).toJSON()).toEqual({ B: 1234 });
+    });
 
-  // We are currently not using this approach, none of the types in Substrate currently
-  // have any overrides. Insteda of trying to support it (just-in-case), rather have it
-  // removed to simplyfy the code - it can be pulled-back if needed
-  it.skip('allows checking against defined indexes', () => {
-    expect(
-      new EnumType(
-        { 1: Text, 5: U32 },
-        new Uint8Array([1, 2 << 2, 49, 50])
-      ).toString()
-    ).toEqual('Text');
-  });
+    // We are currently not using this approach, none of the types in Substrate currently
+    // have any overrides. Instead of trying to support it (just-in-case), rather have it
+    // removed to simplify the code - it can be pulled-back if needed
```

Comment:
Agree with this comment: we could remove _indexes and co in https://github.com/polkadot-js/api/pull/922/files#diff-438804f1fd82048ec30fdbe6c821d298R50

(not urgent)